Fixed Remaining Linting in the rest of repository

iamarshsingh opened the pull request:
Hi, This PR is fixing the remaining of the linting issues in the repository.

The issues that were fixed are as below:
```
************* Module src.python.bot.fuzzers.honggfuzz.engine
src/python/bot/fuzzers/honggfuzz/engine.py:112:45: W0613: Unused argument 'build_dir' (unused-argument)
src/python/bot/fuzzers/honggfuzz/engine.py:182:47: W0613: Unused argument 'arguments' (unused-argument)
************* Module src.python.bot.fuzzers.syzkaller.engine
src/python/bot/fuzzers/syzkaller/engine.py:69:32: W0613: Unused argument 'target_path' (unused-argument)
src/python/bot/fuzzers/syzkaller/engine.py:116:22: W0613: Unused argument 'target_path' (unused-argument)
src/python/bot/fuzzers/syzkaller/engine.py:116:47: W0613: Unused argument 'arguments' (unused-argument)
************* Module src.python.bot.fuzzers.syzkaller.runner
src/python/bot/fuzzers/syzkaller/runner.py:66:38: W0613: Unused argument 'default_args' (unused-argument)
************* Module src.python.tests.core.bot.tasks.impact_task_test
src/python/tests/core/bot/tasks/impact_task_test.py:182:2: C0111: Missing method docstring (missing-docstring)
************* Module src.python.tests.core.platforms.android.sanitizer_test
src/python/tests/core/platforms/android/sanitizer_test.py:70:2: C0111: Missing method docstring (missing-docstring)
src/python/tests/core/platforms/android/sanitizer_test.py:105:2: C0111: Missing method docstring (missing-docstring)
************* Module src.python.bot.minimizer.minimizer
src/python/bot/minimizer/minimizer.py:336:4: W0107: Unnecessary pass statement (unnecessary-pass)
************* Module src.python.bot.fuzzers.engine
src/python/bot/fuzzers/engine.py:25:0: W0622: Redefining built-in 'TimeoutError' (redefined-builtin)
```

Some issues in the code base that I found and have opened a Issue for:
```
************* Module src.python.bot.minimizer.js_minimizer
src/python/bot/minimizer/js_minimizer.py:187:13: E1102: js_minimizer is not callable (not-callable)
src/python/bot/minimizer/js_minimizer.py:188:13: E1102: js_minimizer is not callable (not-callable)
************* Module src.python.bot.minimizer.minimizer
src/python/bot/minimizer/minimizer.py:400:4: E1128: Assigning result of a function call, where the function returns None (assignment-from-none)
```

Some Files that will be deleted in future so their linting issues are not resolved:
```
************* Module src.appengine.appengine_config
************* Module src.python.tests.core.system.path_patcher_test
```

With this almost all of the linting issues have been resolved in the repository so #1824 can be closed after this PR.

Please check and let me know if any changes are required. Thanks!

iamarshsingh commented:
/gcbrun

iamarshsingh commented:
/gcbrun

oliverchang commented on src/python/bot/fuzzers/honggfuzz/engine.py:
```diff
@@ -109,7 +109,8 @@ class HonggfuzzEngine(engine.Engine):
     return 'honggfuzz'
 
-  def prepare(self, corpus_dir, target_path, build_dir):
+  # FIXME: Add support for build_dir argument
```
nit: remove this comment. this argument is optionally handled and doesn't make sense here.

iamarshsingh replied:
yes, sure.

oliverchang commented on src/python/bot/fuzzers/honggfuzz/engine.py:
```diff
@@ -179,7 +180,8 @@ def fuzz(self, target_path, options, reproducers_dir, max_time):
                              stats, fuzz_result.time_executed)
 
-  def reproduce(self, target_path, input_path, arguments, max_time):
+  # FIXME: Add support for additional arguments.
```
same here and in all other cases.

iamarshsingh replied:
sure

oliverchang commented on src/python/bot/untrusted_runner/host.py:
```diff
@@ -160,7 +159,7 @@ def wrapped(*args, **kwargs):
           # Use __str__ to avoid newstrs. TODO(ochang): Use plain str() once
           # migrated to Python 3.
-          raise engine.TimeoutError(e.__str__())
+          raise TimeoutError(e.__str__())
```
can you also replace the `__str__` with a str() call and remove the comment above?

iamarshsingh replied:
oh, sure. Didn't read the comment above.

iamarshsingh commented:
/gcbrun

iamarshsingh commented:
/gcbrun

iamarshsingh commented:
/gcbrun

iamarshsingh commented:
I have also added some Python 2 version check specific code snippets which @oliverchang mentioned in #1824 ([here](https://github.com/google/clusterfuzz/issues/1824#issuecomment-647452823)).

iamarshsingh commented:
/gcbrun

inferno-chromium commented:
Travis is currently failing on 

```
Step #3: Traceback (most recent call last):
Step #3:   File "butler.py", line 316, in <module>
Step #3:     main()
Step #3:   File "butler.py", line 302, in main
Step #3:     command.execute(args)
Step #3:   File "src/local/butler/py_unittest.py", line 283, in execute
Step #3:     run_tests_parallel(args, test_directory, top_level_dir)
Step #3:   File "src/local/butler/py_unittest.py", line 161, in run_tests_parallel
Step #3:     for test_class in subsuite._tests:  # pylint: disable=protected-access
Step #3:   File "/usr/local/lib/python3.7/unittest/loader.py", line 32, in __getattr__
Step #3:     return super(_FailedTest, self).__getattr__(name)
Step #3: AttributeError: 'super' object has no attribute '__getattr__'
Finished Step #3
ERROR
ERROR: build step 3 "gcr.io/clusterfuzz-images/ci" failed: step exited with non-zero status: 1
```

Also, i just merged https://github.com/google/clusterfuzz/pull/1848, please bring those changes in.

iamarshsingh commented:
@inferno-chromium yes, I have fixed those changes. Actually I removed this part of code:

```
# TODO(mbarbella): Remove this and all users after fully migrating to Python 3.
def python2_only(func):
  """Tests which can only run on Python 2."""
  return unittest.skipIf(sys.version_info.major != 2,
                         'Skipping Python 2-only test.')(
                             func)
```

As this is was to be removed after migration to Python3. But I am not really sure why after removing this, python2 tests were also running on the Travis instance due to which it failed. I added it again and the test that failed, I tried to run it locally and it was working. Hopefully it should this time.

I didn't exactly understood why this happens, can you please explain?

iamarshsingh commented:
/gcbrun

inferno-chromium commented on src/local/butler/constants.py:
```diff
@@ -44,14 +44,12 @@
-if sys.version_info.major == 2:
-  ABIS = {'linux': 'cp27mu', 'windows': 'cp27m', 'macos': 'cp27m'}
-elif sys.version_info.major == 3 and sys.version_info.minor == 7:
+if sys.version_info.minor == 7:
```
It is important to have "sys.version_info.major == 3" in both branches for correctness and clarity.

iamarshsingh replied:
yes, sure

inferno-chromium commented on src/local/butler/deploy.py:
```diff
@@ -218,10 +218,8 @@ def _deploy_zip(bucket_name, zip_path):
-    manifest_suffix = '.3'
-  else:
-    manifest_suffix = ''
+  # manifest_suffix refers to majot python version used.
```
I would say leave the deploy.py changes for @oliverchang . we probably need to clear up python3= completely and can come in another cl.

iamarshsingh commented:
/gcbrun

inferno-chromium commented on src/python/base/utils.py:
```diff
@@ -948,21 +947,3 @@ def cpu_count():
-  return future_utils.native(newstr).encode('utf-8')
-
-
-def exc_clear():
```
This is not removed from here.

```
src/python/bot/startup/heartbeat.py:        utils.exc_clear()
src/python/bot/startup/run_bot.py:    utils.exc_clear()
src/python/bot/tasks/commands.py:  utils.exc_clear()
```

iamarshsingh replied:
I have removed it from all required places.

inferno-chromium commented on src/python/base/utils.py:
```diff
@@ -948,21 +947,3 @@ def cpu_count():
-
-
-# TODO(mbarbella): Delete this once fully migrated to Python 3.
-def newstr_to_native_str(s):
```
Make sure to git grep "newstr_to_native_str" to make sure you removed all, i see some, just want to make sure.

inferno-chromium commented on src/python/bot/fuzzers/libFuzzer/engine.py:
```diff
@@ -558,7 +556,7 @@ def cleanse(self, target_path, arguments, input_path, output_path, max_time):
         additional_args=arguments)
 
     if result.timed_out:
-      raise engine.TimeoutError('Cleanse timed out\n' + result.output)
```
git grep "engine.TimeoutError" just to make sure all instances are converted.

inferno-chromium commented on src/python/bot/minimizer/minimizer.py:
```diff
@@ -397,6 +396,7 @@ def _attempt_merge(self, hypotheses, sibling_merge_succeeded=False):
     back = hypotheses[middle:]
 
     # If we could remove either one of two hypotheses, favor removing the first.
+    # pylint: disable=assignment-from-none
```
What is this ? @mbarbella-chromium fyi

iamarshsingh replied:
This I have to add because self._attempt_merge doesn't returns anything but according to the logic of the code, this should returns something. This is probably a bug, I confirmed with @oliverchang and have reported it here #1845

inferno-chromium replied:
Probably better to not add this pylint then as it is a real bug.

iamarshsingh replied:
If I removed this pyllint, the travis CI build will fail as I have made another change to this file. I can probably remove both of them and after fixing this bug we can change the other lint issue?

inferno-chromium replied:
ok fine to keep then. add another: 
FIXME: Fix this. Tracked in https://github.com/google/clusterfuzz/issues/1845.

inferno-chromium commented on src/python/bot/tasks/update_task.py:
```diff
@@ -46,9 +46,7 @@
-MANIFEST_FILENAME = 'clusterfuzz-source.manifest'
-if sys.version_info.major == 3:
-  MANIFEST_FILENAME += '.3'
+MANIFEST_FILENAME = 'clusterfuzz-source.manifest.3'
```
Lets remove these changes, will let @oliverchang do these with deploy.py, these can break things.

inferno-chromium commented on src/python/bot/untrusted_runner/host.py:
```diff
@@ -157,10 +156,7 @@ def wrapped(*args, **kwargs):
-          # Use __str__ to avoid newstrs. TODO(ochang): Use plain str() once
-          # migrated to Python 3.
-          raise engine.TimeoutError(e.__str__())
+          raise TimeoutError(e.str())
```
did you mean str(e) ? @oliverchang to review.

iamarshsingh replied:
This is left for review.

inferno-chromium replied:
I think it is str(e), lets do and then @oliverchang can do final review and land.

inferno-chromium commented on src/python/chrome/crash_uploader.py:
```diff
@@ -339,10 +338,7 @@ def parse_mime_to_crash_report_info(local_minidump_mime_path):
-    minidump_mime_contents = email.message_from_bytes(minidump_mime_bytes)
-  else:
-    minidump_mime_contents = email.message_from_string(minidump_mime_bytes)
+  minidump_mime_contents = email.message_from_string(minidump_mime_bytes)
```
email.message_from_bytes, looks like wrong subsitution

iamarshsingh replied:
yes, thank you for pointing this out.

inferno-chromium commented on src/python/system/shell.py:
```diff
@@ -369,12 +369,6 @@ def get_interpreter(file_to_execute, is_blackbox_fuzzer=False):
   except KeyError:
     return None
 
-  # TODO(mbarbella): Remove this when fuzzers have been migrated to Python 3.
```
Need to keep this if. this is currently used, and then above pylint can be removed.

inferno-chromium commented on src/python/tests/core/bot/tasks/update_task_test.py:
```diff
@@ -13,7 +13,6 @@
 # limitations under the License.
 """update_task tests."""
```
Probably revert these update_task_test, these need to fixed with deployment changes in another Cl by @oliverchang

inferno-chromium commented on src/python/tests/core/local/butler/deploy_test.py:
```diff
@@ -17,7 +17,6 @@
 import json
```
Probably revert these deploy_test, these need to fixed with deployment changes in another Cl by @oliverchang

inferno-chromium commented on src/local/butler/guard.py:
```diff
@@ -13,14 +13,10 @@
 
 def check_virtualenv():
   """Check that we're in a virtualenv."""
-  if sys.version_info.major != 3:
```
This is important to keep!

inferno-chromium commented on src/local/butler/py_unittest.py:
```diff
@@ -168,7 +158,6 @@ def run_tests_parallel(args, test_directory, top_level_dir):
       if subsuite.__class__.__name__ == 'ModuleImportFailure':
         unittest.TextTestRunner(verbosity=1).run(subsuite)
         raise Exception('A failure occurred while importing the module.')
-
```
lets keep this.

iamarshsingh commented:
/gcbrun

iamarshsingh commented:
/gcbrun

iamarshsingh commented:
/gcbrun

inferno-chromium commented on src/python/system/shell.py:
```diff
@@ -369,7 +369,6 @@ def get_interpreter(file_to_execute, is_blackbox_fuzzer=False):
   except KeyError:
     return None
 
-  # TODO(mbarbella): Remove this when fuzzers have been migrated to Python 3.
```
Don't remove the TODO.

inferno-chromium commented on src/python/system/shell.py:
```diff
@@ -352,6 +351,7 @@ def get_free_disk_space(path='/'):
   return psutil.disk_usage(path).free
 
 
+# pylint: disable=unused-argument
```
Remove this pylint, i think we are using is_blackbox_fuzzer now.

iamarshsingh commented:
/gcbrun

oliverchang commented on src/python/bot/untrusted_runner/untrusted.py:
```diff
@@ -215,9 +215,7 @@ def _get_tls_cert_and_key():
   # TODO(mbarbella): Remove this after migrating to Python 3. The grpc library
   # has explicit type checks against str.
   cert_contents = str(compute_metadata.get('instance/attributes/tls-cert'))
-  cert_contents = utils.newstr_to_native_str(cert_contents)
```
This changes the type from "bytes" to "str. Can you fix this such that it looks like:

```python
cert_contents = compute_metadata.get('instance/attributes/tls-cert').encode()
...
```

I believe the `str` is no longer necessary either.

iamarshsingh replied:
Thanks, Updated in the PR. Please check!

inferno-chromium commented on src/python/bot/untrusted_runner/host.py:
```diff
@@ -157,10 +156,7 @@ def wrapped(*args, **kwargs):
-          # Use __str__ to avoid newstrs. TODO(ochang): Use plain str() once
-          # migrated to Python 3.
-          raise engine.TimeoutError(e.__str__())
+          raise TimeoutError(e)
```
I think this should be TimeoutError(str(e)) to avoid any unexpected issues with implicit string conversion.

iamarshsingh replied:
Thanks, updated in the PR.

iamarshsingh commented:
/gcbrun

iamarshsingh commented:
/gcbrun
